kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/ExecListener.java

```diff
@@ -15,32 +15,38 @@
  */
 package io.fabric8.kubernetes.client.dsl;
 
-import okhttp3.Response;
-
+import java.io.IOException;
 
 public interface ExecListener {
+  
+  public interface Response {
+    
+    int code();
+    
+    /**
+     * May be null if not provided by the underlying implementation.
+     * @return the body as a String
+     * @throws IOException
+     */
+    String body() throws IOException;
+    
+  }
 
   /**
    * Called when the request has successfully been upgraded to a web socket.
-   *
-   * @param response OkHttp response object
    */
-  void onOpen( Response response);
-
-    /**
-     * Called when the transport or protocol layer of this web socket errors during communication.
-     *
-     * @param t Throwable
-     * @param response Present when the failure is a direct result of the response (e.g., failed
-     * upgrade, non-101 response code, etc.). {@code null} otherwise.
-     */
-    void onFailure(Throwable t, Response response);
+  default void onOpen() {}
 
+  /**
+   * Called when the transport or protocol layer of this web socket errors during communication.
+   *
+   * @param t Throwable
+   * @param failureResponse non-null if the failure is caused by the handshake
+   */
+  default void onFailure(Throwable t, Response failureResponse) {}
```

Comment:
Shouldn't the second parameter (`Response failureResponse`) be removed here?

Reply:
Or replaced by the new HttpResponse, or  response code + message

Reply:
This is a local ExecListener.Response that contains just the code and body (which should be available on later java versions with the jdk client)

Reply:
:flushed: I missed the line where it was defined. I knew we discussed this internally.